standards/core/mix.md

```diff
@@ -0,0 +1,128 @@
+A node that initiates the request is a `service-user/client` whereas a node that replies to the request is the `service-provider/service-node`.
+
+This document also covers the aspect of relay nodes acting as mix nodes.
+Integration of [waku relay](https://rfc.vac.dev/waku/standards/core/11/relay) with mix is not covered in this document.
+
+## Background / Rationale / Motivation
+
+Waku protocols have weak sender/originator anonymity as explained in [Waku Privacy and Anonymity Analysis](https://vac.dev/rlog/wakuv2-relay-anon/).
+Without further anonymization, it is easy to analyze the network traffic and determine the originator of messages published into the network.
+Topic interests of a user can be identified by analyzing [Store](https://rfc.vac.dev/waku/standards/core/13/store) query messages sent by a user.
+Same applies for topic subscriptions via [Filter](https://rfc.vac.dev/waku/standards/core/12/filter) protocol.
+
+`Mix` protocol allows libp2p nodes to send messages without revealing the sender's identity (peer ID, IP address) to intermediary mix nodes or the recipient/destination.
+Anonymity is achieved by using the [Sphinx packet format](#references), which encrypts and routes messages through a series of mix nodes before reaching the recipient.
+
+By integrating the `mix` protocol into the waku network, we can improve the anonymity for publishers and store query users.
+Each waku relay node will be acting as a mix node that forms an `overlay mix network`.
+This network of mix nodes SHALL relay mix messages anonymously to the recepient.
+
+Anonymity of [Filter](https://rfc.vac.dev/waku/standards/core/12/filter) users is not addressed by this document.
+
+## Terminology
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”,
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Theory / Semantics
+
+Waku Mix creates an `overlay network` of all the waku nodes that support the `mix` protocol.
+
+Nodes with `mix` protocol mounted SHOULD advertise that they support mix protocol via their their chosen discovery method.
+They MAY do so by updating their [ENR](#enr-updates) and using one of the ENR based discovery methods.
+
+Nodes that want higher anonymity while publishing a message via [lightpush](https://rfc.vac.dev/waku/standards/core/19/lightpush) or performing a [store](https://rfc.vac.dev/waku/standards/core/13/store) query SHOULD use the `mix` protocol to route their messages to the destination.
+Sender Node that would like to use `mix` protocol SHOULD discover enough mix nodes so that there is always a healthy pool of mix nodes available for selection.
+The pool size of mix nodes SHOULD be large enough for the mixing to be effective.
+We RECOMMEND a pool size of at least 100 mix nodes for the mixing to be effective.
```

Comment:
Is there any calculation to back up this number? What is considered "effective", by what metric?